### ISSUE TITLE
Rename Hidden Combo Button ''B+X to A+B''

### ISSUE THREAD

**Issue by rslkr** [yoshisuga/MAME4iOS#159]
The main buttons used to be ''X and B'' on screen controller.
I understand and respect the idea that why the buttons were changed to ''A and B''.
They are adjustable but I still don't get used to it. It's kinda annoys me.

I'll explain it more in e.g. below.
I use left hand's thumb for j.stick and right hand's thumb for buttons.
Sorry for the mere detail and yes I'm not a freak :)
Some games require 2 buttons for special moves and I was using the hidden button ''B+X'' for it.
I can't touch on 2 buttons same time even they next to each other! It was good before now I have to adjust more things.

In short, please someone can add one more hidden button for ''A+B''
And these hidden buttons could turn on/off individually in the settings by user.
If the user does not want them to be active on screen controller.
Like ''Fullscreen Buttons''

**Comment by ToddLa** [comment]
![image](https://user-images.githubusercontent.com/4494698/78974803-a5628800-7ac7-11ea-9e56-4c93f88fd77d.jpeg)

If you go into Settings > Game Input > Change Current layout you’re can see the locations of the hidden buttons. 

Is it just placed wrong? Missing? Or something else.

**Comment by ToddLa** [comment]
The default button order should be ABYXLR, so what you are saying is we need a A+B touch area?

**Comment by rslkr** [comment]
Yes exactly.

**Comment by rslkr** [comment]
I'm playing the Final Fight so I can't do special moves because main buttons are right now 
''A and B'' so there is no hidden ''A+B'' button

**Comment by rslkr** [comment]
BTW I forgot to thank you for fixing the settings icon now looks better.

**Comment by ToddLa** [comment]
It would be a PITA to add a new combo button, Ana do layout for every device!

Would it make sense to “rename” one of the existing combo buttons to A+B?

**Comment by ToddLa** [comment]
You could also go into the MAME menu, and make A and X the first, second buttons for this game, then the existing A+X ghost button would work for you?

**Comment by rslkr** [comment]
> It would be a PITA to add a new combo button, Ana do layout for every device!

I understand. I thought it would be easy to make one more combo.

> Would it make sense to “rename” one of the existing combo buttons to A+B?

Good idea. This would work for me
But and then some other user may start complain about this...
They may open a issue for that's b'coz of this...

> You could also go into the MAME menu, and make A and X the first, second buttons for this game, then the existing A+X ghost button would work for you?

The existing A+X ghost button! What is this mean?
This is what I do since the button ''X'' replaced to the button ''A''
(depends on the game) Most of times I use ''Auto'' in Settings>Input Option>Fullscreen Buttons
e.g. for Final Fight it shows 3 buttons Y A and B (Fullscreen) but I need ''X'' for combo B+X button
Here is the annoying part begins.
So I select 4 buttons in the settings for button X and also remap the buttons in MAME input.
2 buttons enough for this game. One for hit one for jump. And the combo of course. 

And please forget about the joystick, that was a ridiculous request. Sorry.

**Comment by rslkr** [comment]
For 6 buttons games on iPhone7 I don't need to change layout.
The buttons layout fit exactly where its belongs to. Like classic arcade machine.  

![IMG_1222](https://user-images.githubusercontent.com/46307397/79295613-6d6ea100-7f1c-11ea-838f-2e711c560f86.PNG)

But for iPhoneX I need to change layout.That's why mostly I use iPhone7.

![8275b482-7738-4be7-bbd7-fb75974299b2](https://user-images.githubusercontent.com/46307397/79296249-3ac5a800-7f1e-11ea-97ef-93ac5e8878d6.JPG)

For 2 buttons games on iPhone7 I still need little adjustment.Since the button ''X'' replaced to the button ''A''
Before I did not need to change buttons layout.The main active buttons were next to each other
(B and X)

![IMG_1234](https://user-images.githubusercontent.com/46307397/79297654-1cfa4200-7f22-11ea-841f-2f28ca99624d.PNG)

**Comment by ToddLa** [comment]
I would think a layout like this one....


https://user-images.githubusercontent.com/46307397/79296249-3ac5a800-7f1e-11ea-97ef-93ac5e8878d6.JPG


Would be the standard, I like the L1,R1 buttons on the left and the right, where they belong. And the ABYX in the lower right, and the stick in the lower left.

**Comment by rslkr** [comment]
I wish that ''L'' button could be right side as well but doesn't matter.I can remap the button that I use less.Its all good.
Thank you.

**Comment by rslkr** [comment]
Hi @ToddLa Its been a while and no one has said nothing so far.

As you recommended before, I think better way to do this, rename one of the existing combo buttons. 

We can rename the combo ''B+X'' to ''A+B''
And also B+X is on the right place.

I noticed this:
In Game Input> Fullscreen Stick Size not working anymore!
And in Game Input> Button Size is changing sizes both of Buttons and Stick together!

**Comment by ToddLa** [comment]
![image](https://user-images.githubusercontent.com/4494698/81566211-68e8ad00-934f-11ea-9e0c-625e7ee829b9.jpeg)

This is what the hidden buttons are like now, after having all layouts follow consistent 🅐 🅑 🅧 🅨 placement.

**Comment by ToddLa** [comment]
🅐🅧  and 🅑🅨 have a sane combo, but not the other two

**Comment by ToddLa** [comment]
Rename/Change B+X to A+B and A+Y to X+Y???

I don’t know the original  need for these combos, and if they should exist at all??

**Comment by rslkr** [comment]
The rest of combo buttons I think they may useful like for King Of Fighters games for some special moves.
I have to look the some games.

But B+X (A+B) this definitely requires for 2 buttons game.

**Comment by rslkr** [comment]
I just check the KOF95.
Y+B No need to change. It works for reduce Opponent's POW energy.
A+B also works for this game. Sidesteps moves.

**Comment by rslkr** [comment]
According to this KOF95 :
We need to rename and change 2 combo buttons.

- B+X to A+B

- X+A to B+X

A+B very useful for:

- Cadillacs and Dinosaurs

- Final Fight

- Punisher

**Comment by ToddLa** [comment]
I added a `A+B` button to PR #243

**Comment by rslkr** [comment]
Thank you ToddLa.
The position of A+B should be replaced with B+X

> Should I add X+Y too?

Honestly, I only use and need A+B
And I'm also wondering Why David (Seleuco) specifically picked those combos?
But one thing for sure, he was doing good stuff on this MFi back in time 'till Apple pulled his Gridlee on App Store. 
The combos one of them. **The unique features.**
You are doing the same thing, keep it up sir. 👍

**Comment by rslkr** [comment]
Thank you, ToddLa.
Close this issue.
